### PR TITLE
Changing messaging to Clayton's spec

### PR DESCRIPTION
**Pull request by jupierce**
Attempting to address concerns from openshift/origin#9247

### CODE REVIEW

File: pkg/docker/docker.go [hunk 1]
```diff
@@ -347,16 +347,26 @@ func (d *stiDocker) GetOnBuild(name string) ([]string, error) {
 // and returns the image metadata
 func (d *stiDocker) CheckAndPullImage(name string) (*docker.Image, error) {
 	name = getImageName(name)
+	displayName := name
+
+	if glog.Is(3) == false {
```
Comment (php-coder) [inline pkg/docker/docker.go]: Why we don't need this on level 3? Looks like you've tried to write condition like `if logLevel < 3`. If it's true I see that you missed levels 4 and 5.

Comment (php-coder) [inline pkg/docker/docker.go]: BTW, what is "Clayton's spec" and where I can find it?

Comment (csrwng) [inline pkg/docker/docker.go]: @php-coder It's comments on this issue: https://github.com/openshift/origin/issues/9247

glog.Is(x) really means loglevel >= x ... not very well named, but it's used correctly above.

@jupierce why not  if !glog.Is(3) ?

**Comment by csrwng** [comment]
We really need to use the imageprogress pull writer as used here:
https://github.com/openshift/origin/blob/master/pkg/util/docker/dockerfile/builder/client.go#L326-L327

We should add an option to the s2i builder config to specify the output writer and raw flag for image pulls and make it available here.

File: pkg/docker/docker.go (hunk 1 above)
Comment (php-coder) [inline pkg/docker/docker.go]: @csrwng Thank you for explanation and a link. I was out of context but now it's clear.

File: pkg/docker/docker.go [hunk 2]
```diff
@@ -347,16 +347,26 @@ func (d *stiDocker) GetOnBuild(name string) ([]string, error) {
...
+		// Summarize name for lower level log output
+		split := strings.Split(name, "@")
+		if len(split) > 1 && len(split[1]) > 18 {
```
Comment (csrwng) [inline pkg/docker/docker.go]: What's the purpose for checking len > 18 ?

Comment (csrwng) [inline pkg/docker/docker.go]: btw, there's a utility function here that parses the ref here:
https://github.com/openshift/origin/blob/master/pkg/image/api/helper.go#L133

We can't use it from here but maybe we can take the useful bits.

File: pkg/docker/docker.go [hunk 3]
```diff
@@ -347,16 +347,26 @@ func (d *stiDocker) GetOnBuild(name string) ([]string, error) {
...
+			displayName = split[0] + "@" + split[1][:18] + "..."
```
Comment (bparees) [inline pkg/docker/docker.go]: i assume 18 refers to the id length, but a comment would be good.

**Comment by bparees** [comment]
@jupierce can we get a sample of the default output?

File: pkg/docker/docker.go (hunk 1 above)
Comment (jupierce) [inline pkg/docker/docker.go]: @csrwng  Will change to !glog in next commit

File: pkg/docker/docker.go (hunk 3 above)
Comment (jupierce) [inline pkg/docker/docker.go]: @csrwng  @bparees  Clayton suggested a "summary" at this logging level. The choice of 18 was somewhat arbitrary: I wanted the user to see enough of the sha for it to be unlikely to correspond to existing images on the system, but shorter than the 80 chars Clayton believed was problematic.
So it changes:
**Image "centos/php-56-centos7@sha256:51c3e2b08bd9fadefccd6ec42288680d6d7f861bdbfbd2d8d24960621e4e27f5" not available locally, pulling ...**
to 
**Image "centos/php-56-centos7@sha256:51c3e2b08bd..." not available locally, pulling ...**

**Comment by jupierce** [comment]
Working on resucitating my golang environment and I will provide an actual log.

File: pkg/docker/docker.go (hunk 3 above)
Comment (bparees) [inline pkg/docker/docker.go]: i see.  even more reason for a comment in the code then.

Comment (smarterclayton) [inline pkg/docker/docker.go]: Note that docker also shortens these in some contexts

**Comment by jupierce** [comment]
s2i runs at all log levels:
https://gist.github.com/jupierce/59413ff40e89127b5b1b9f16c9c62aa4

File: pkg/scripts/install.go [hunk 4]
```diff
@@ -120,7 +120,7 @@ func (s *SourceScriptHandler) Get(script string) *api.InstallResult {
 	// and this should (and will) be removed soon.
 	location = strings.Replace(location, "s2i/bin", "sti/bin", 1)
 	if s.fs.Exists(location) {
-		glog.Info("DEPRECATED: Use .s2i/bin instead of .sti/bin")
+		glog.V(3).Infof("DEPRECATED: Use .s2i/bin instead of .sti/bin")
```
Comment (bparees) [inline pkg/scripts/install.go]: this one can stay, people should be told if they're using deprecated logic.

Comment (jupierce) [inline pkg/scripts/install.go]: Restored the deprecated output to use glog.Info.

**Comment by jupierce** [comment]
Changed downloading and setting environment variables to V(1) per @bparees .

**Comment by bparees** [comment]
lgtm [merge]

**Pull request merged**